### PR TITLE
PL 65 user search functionality

### PR DESCRIPTION
**Pull request by samsawan**
# Things Done
- utilize [Fuse library](http://fusejs.io/) for fuzzy searching of users
- robust user search behavior
  - remembers what users were selected and prepends them to top of user list when search is retriggered
- move all event handlers to `widget.js`
- build size is 282K

# How To Test
- spin up the widget
- start searching

### CODE REVIEW

File: src/js/widget.js [hunk 1]
```diff
@@ -379,21 +389,59 @@ class SBWidget {
...
+                this.chatSection.addClickEvent(userItem, () => {
+                    flipClass(userItem.select, className.ACTIVE);
+
+                    let selectedUserCount = this.chatSection.getSelectedUserIds(userContent.list).length;
+                    this.chatSection.updateChatTop(target, selectedUserCount > 9 ? MAX_COUNT : selectedUserCount.toString(), null);
+                    selectedUserCount > 0 ? removeClass(target.startBtn, className.DISABLED) : addClass(target.startBtn, className.DISABLED);
+                });
+            };
+
+            for (let i = 0 ; i < activeUsers.length ; i++) {
```
Comment (gregsvo) [inline src/js/widget.js]: Wrote the following, but then saw that you are using map on line 429.

This seems like a good place to use .map() as it seems like a more functional approach to me. But no big deal.
```
activeUsers.map(doThing)

let doThing = (user) => {
    let item = this.chatSection.createUserListItem(user, true);
    clickEvent(item);
    userContent.list.appendChild(item);
    }
```

Comment (samsawan) [reply src/js/widget.js]: The line 429 map is just a quick JSON parser over an array of JSON so I felt okay-ish writing it. I'm gonna say upwards of 95% of all the loops in this thing are the old style `for` loops, so I'm gonna keep the meatier loops that way. 

I do see an opportunity for a bit of DRYness with your rewrite: the three lines in the `doThing` function are repeated when I render the inactiveUser list. I'll do that refactoring.

File: src/js/widget.js [hunk 2]
```diff
@@ -525,22 +573,71 @@ class SBWidget {
...
+                    if(!activeUserIds.includes(user.userId)) {
+                        let item = this.popup.createMemberItem(user);
+                        this.popup.addClickEvent(item, clickEvent(item));
+                        this.popup.invitePopup.list.appendChild(item);
+                    }
+                }
+            };
+
+            let seperateAndClearUserList = (userList) => {
+                let activeSelection = (user) => {
```
Comment (hafley66) [inline src/js/widget.js]: this is a pure function, lift to top level scope

Comment (samsawan) [reply src/js/widget.js]: Not gonna die on this hill, but do you always lift pure functions to the top level? I thought by scoping it to the `seperateAndClearUserList` function I'm making it clear that the `activeSelection` function is only used by the `seperateAndClearUserList`. I'm new to JS so I'm not fighting about it and I don't have strong opinions.

**Pull request merged**